x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -5,21 +5,36 @@
  * 2.0.
  */
 
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useReducer } from 'react';
 import { merge } from 'lodash';
-import { CasesContextValue, CasesFeatures } from '../../../common/ui/types';
 import { DEFAULT_FEATURES } from '../../../common/constants';
 import { DEFAULT_BASE_PATH } from '../../common/navigation';
 import { useApplication } from './use_application';
+import {
+  CasesContextStoreAction,
+  casesContextReducer,
+  getInitialCasesContextState,
+} from './cases_context_reducer';
+import { CasesContextFeatures, CasesFeatures } from '../../containers/types';
+import { CasesGlobalComponents } from './cases_global_components';
 
-export const CasesContext = React.createContext<CasesContextValue | undefined>(undefined);
+export interface CasesContextValue {
+  owner: string[];
+  appId: string;
+  appTitle: string;
+  userCanCrud: boolean;
+  basePath: string;
+  features: CasesContextFeatures;
+  dispatch: (action: CasesContextStoreAction) => void;
```

Comment:
I think you should use one of the `React`s types for that. Something like, `dispatch: Dispatch<ReducerAction<CasesContextStoreAction>>`.

Reply:
I haven't use those types. I'll check it out.

Reply:
I added the Dispatch type thanks! I won't use ReducerAction since I am already typing the actions.